servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/RequestHeadersPropagatorSetter.java

```diff
@@ -31,7 +30,7 @@ private RequestHeadersPropagatorSetter() {
     @Override
     public void set(@Nullable final RequestInfo requestInfo, final String key, final String value) {
         if (requestInfo != null) {
-            requestInfo.request().headers().set(key.toLowerCase(Locale.ENGLISH), value);
+            requestInfo.request().headers().set(key, value);
```

Comment:
Continuation of https://github.com/apple/servicetalk/pull/3306#discussion_r2241556810

I'm all for for fixing that at the root, but just bcz we currently depend on their behavior and timeline for fixing it there is unclear, imho we should keep `toLowerCase` as it was here before. Otherwise, you risk breaking existing users and it won't work with gRPC & B3 for your next PR.

Reply:
Okay, let's keep the lower case for now.